launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -311,11 +311,17 @@ private List<String> buildSparkSubmitCommand(Map<String, String> env)
   }
 
   private void checkJavaOptions(String javaOptions) {
-    if (!isEmpty(javaOptions) && javaOptions.contains("Xmx")) {
-      String msg = String.format("Not allowed to specify max heap(Xmx) memory settings through " +
-        "java options (was %s). Use the corresponding --driver-memory or " +
-        "spark.driver.memory configuration instead.", javaOptions);
-      throw new IllegalArgumentException(msg);
+    System.out.println(javaOptions);
+
+    if (!isEmpty(javaOptions)) {
+      for (String javaOption: javaOptions.split(" ")) {
```

Comment:
This is not robust - you can have space within the value of a property by quoting it.
```
$ scala -Dprop="abc 123"
Welcome to Scala 2.12.14 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_212).
Type in expressions for evaluation. Or try :help.

scala> System.getProperty("prop")
res0: String = abc 123
```

Reply:
Maybe `org.apache.spark.launcher.CommandBuilderUtils#parseOptionString` is ok?

Reply:
Perfect @LuciferYang  !

Reply:
Ah I missed your comment and had reimplemented a kind of parser...
Thks for the pointer I will re-update the code to rely on this one